From b84b3dc0ea7d1fb15a50fdcbe47f4beb195748a8 Mon Sep 17 00:00:00 2001
From: Ilya Ledvich <ilya@compulab.co.il>
Date: Sun, 17 Mar 2024 13:43:13 +0200
Subject: [PATCH 082/155] compulab: dts: imx8plus: correct pad settings

According to SOC's TRM bit layout, BIT3 and BIT0 are reserved:
 8  7   6   5   4   3  2 1  0
PE HYS PUE ODE FSEL X  DSE  X

This patch corrects pad settings for C-Lab products based on imx8plus SOC.
Sync with compulab-linux.

Signed-off-by: Ilya Ledvich <ilya@compulab.co.il>
---
 arch/arm/dts/mcm-imx8m-plus.dts          |  52 ++++-----
 arch/arm/dts/sb-iotgimx8plus.dtsi        |   6 +-
 arch/arm/dts/som-imx8m-plus.dts          | 128 +++++++++++------------
 arch/arm/dts/ucm-imx8m-plus-pinctrl.dtsi | 102 +++++++++---------
 arch/arm/dts/ucm-imx8m-plus.dts          |  86 +++++++--------
 5 files changed, 187 insertions(+), 187 deletions(-)

diff --git a/arch/arm/dts/mcm-imx8m-plus.dts b/arch/arm/dts/mcm-imx8m-plus.dts
index cfadc3761f7..02a5ee71f2b 100644
--- a/arch/arm/dts/mcm-imx8m-plus.dts
+++ b/arch/arm/dts/mcm-imx8m-plus.dts
@@ -576,69 +576,69 @@
 
 	pinctrl_gpio_led: gpioledgrp {
 		fsl,pins = <
-			MX8MP_IOMUXC_GPIO1_IO12__GPIO1_IO12	0x19
+			MX8MP_IOMUXC_GPIO1_IO12__GPIO1_IO12	0x140
 		>;
 	};
 
 	pinctrl_i2c1: i2c1grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C1_SCL__I2C1_SCL		0x400001c3
-			MX8MP_IOMUXC_I2C1_SDA__I2C1_SDA		0x400001c3
+			MX8MP_IOMUXC_I2C1_SCL__I2C1_SCL		0x400001c2
+			MX8MP_IOMUXC_I2C1_SDA__I2C1_SDA		0x400001c2
 		>;
 	};
 
 	pinctrl_i2c2: i2c2grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C2_SCL__I2C2_SCL			0x400001c3
-			MX8MP_IOMUXC_I2C2_SDA__I2C2_SDA			0x400001c3
+			MX8MP_IOMUXC_I2C2_SCL__I2C2_SCL			0x400001c2
+			MX8MP_IOMUXC_I2C2_SDA__I2C2_SDA			0x400001c2
 		>;
 	};
 
 	pinctrl_i2c3: i2c3grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C3_SCL__I2C3_SCL			0x400001c3
-			MX8MP_IOMUXC_I2C3_SDA__I2C3_SDA			0x400001c3
+			MX8MP_IOMUXC_I2C3_SCL__I2C3_SCL			0x400001c2
+			MX8MP_IOMUXC_I2C3_SDA__I2C3_SDA			0x400001c2
 		>;
 	};
 
 	pinctrl_i2c5: i2c5grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_SAI5_MCLK__I2C5_SDA		0x400001c3
-			MX8MP_IOMUXC_SAI5_RXD0__I2C5_SCL		0x400001c3
+			MX8MP_IOMUXC_SAI5_MCLK__I2C5_SDA		0x400001c2
+			MX8MP_IOMUXC_SAI5_RXD0__I2C5_SCL		0x400001c2
 		>;
 	};
 
 	pinctrl_i2c1_gpio: i2c1grp-gpio {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C1_SCL__GPIO5_IO14        	0x1c3
-			MX8MP_IOMUXC_I2C1_SDA__GPIO5_IO15        	0x1c3
+			MX8MP_IOMUXC_I2C1_SCL__GPIO5_IO14        	0x1c2
+			MX8MP_IOMUXC_I2C1_SDA__GPIO5_IO15        	0x1c2
 		>;
 	};
 
 	pinctrl_i2c2_gpio: i2c2grp-gpio {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C2_SCL__GPIO5_IO16        	0x1c3
-			MX8MP_IOMUXC_I2C2_SDA__GPIO5_IO17        	0x1c3
+			MX8MP_IOMUXC_I2C2_SCL__GPIO5_IO16        	0x1c2
+			MX8MP_IOMUXC_I2C2_SDA__GPIO5_IO17        	0x1c2
 		>;
 	};
 
 	pinctrl_i2c3_gpio: i2c3grp-gpio {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C3_SCL__GPIO5_IO18        	0x1c3
-			MX8MP_IOMUXC_I2C3_SDA__GPIO5_IO19        	0x1c3
+			MX8MP_IOMUXC_I2C3_SCL__GPIO5_IO18        	0x1c2
+			MX8MP_IOMUXC_I2C3_SDA__GPIO5_IO19        	0x1c2
 		>;
 	};
 
 	pinctrl_i2c5_gpio: i2c5grp-gpio {
 		fsl,pins = <
-			MX8MP_IOMUXC_SAI5_MCLK__GPIO3_IO25       	0x1c3
-			MX8MP_IOMUXC_SAI5_RXD0__GPIO3_IO21        	0x1c3
+			MX8MP_IOMUXC_SAI5_MCLK__GPIO3_IO25       	0x1c2
+			MX8MP_IOMUXC_SAI5_RXD0__GPIO3_IO21        	0x1c2
 		>;
 	};
 
 	pinctrl_pmic: pmicirq {
 		fsl,pins = <
-			MX8MP_IOMUXC_GPIO1_IO03__GPIO1_IO03	0x41
+			MX8MP_IOMUXC_GPIO1_IO03__GPIO1_IO03	0x000001c0
 		>;
 	};
 
@@ -656,7 +656,7 @@
 
 	pinctrl_reg_usdhc2_vmmc: regusdhc2vmmc {
 		fsl,pins = <
-			MX8MP_IOMUXC_SD2_RESET_B__GPIO2_IO19	0x41
+			MX8MP_IOMUXC_SD2_RESET_B__GPIO2_IO19	0x140
 		>;
 	};
 
@@ -668,15 +668,15 @@
 
 	pinctrl_uart1: uart1grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_SD1_CMD__UART1_DCE_RX	0x49
-			MX8MP_IOMUXC_SAI2_RXFS__UART1_DCE_TX	0x49
+			MX8MP_IOMUXC_SD1_CMD__UART1_DCE_RX	0x140
+			MX8MP_IOMUXC_SAI2_RXFS__UART1_DCE_TX	0x140
 		>;
 	};
 
 	pinctrl_uart2: uart2grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_UART2_RXD__UART2_DCE_RX	0x49
-			MX8MP_IOMUXC_UART2_TXD__UART2_DCE_TX	0x49
+			MX8MP_IOMUXC_UART2_RXD__UART2_DCE_RX	0x140
+			MX8MP_IOMUXC_UART2_TXD__UART2_DCE_TX	0x140
 		>;
 	};
 
@@ -694,7 +694,7 @@
 			MX8MP_IOMUXC_SD2_DATA1__USDHC2_DATA1	0x1d0
 			MX8MP_IOMUXC_SD2_DATA2__USDHC2_DATA2	0x1d0
 			MX8MP_IOMUXC_SD2_DATA3__USDHC2_DATA3	0x1d0
-			MX8MP_IOMUXC_GPIO1_IO04__USDHC2_VSELECT	0xc1
+			MX8MP_IOMUXC_GPIO1_IO04__USDHC2_VSELECT	0xc0
 		>;
 	};
 
@@ -706,7 +706,7 @@
 			MX8MP_IOMUXC_SD2_DATA1__USDHC2_DATA1	0x1d4
 			MX8MP_IOMUXC_SD2_DATA2__USDHC2_DATA2	0x1d4
 			MX8MP_IOMUXC_SD2_DATA3__USDHC2_DATA3	0x1d4
-			MX8MP_IOMUXC_GPIO1_IO04__USDHC2_VSELECT 0xc1
+			MX8MP_IOMUXC_GPIO1_IO04__USDHC2_VSELECT 0xc0
 		>;
 	};
 
@@ -718,7 +718,7 @@
 			MX8MP_IOMUXC_SD2_DATA1__USDHC2_DATA1	0x1d6
 			MX8MP_IOMUXC_SD2_DATA2__USDHC2_DATA2	0x1d6
 			MX8MP_IOMUXC_SD2_DATA3__USDHC2_DATA3	0x1d6
-			MX8MP_IOMUXC_GPIO1_IO04__USDHC2_VSELECT 0xc1
+			MX8MP_IOMUXC_GPIO1_IO04__USDHC2_VSELECT 0xc0
 		>;
 	};
 
diff --git a/arch/arm/dts/sb-iotgimx8plus.dtsi b/arch/arm/dts/sb-iotgimx8plus.dtsi
index 0cea893c483..c7090a30f9c 100644
--- a/arch/arm/dts/sb-iotgimx8plus.dtsi
+++ b/arch/arm/dts/sb-iotgimx8plus.dtsi
@@ -125,13 +125,13 @@
 &iomuxc {
 	pinctrl_fec_phy_reset: fecphyrstgrp {
 		fsl,pins = <
-			MX8MP_IOMUXC_SAI2_TXFS__GPIO4_IO24			0x19
+			MX8MP_IOMUXC_SAI2_TXFS__GPIO4_IO24			0x10
 		>;
 	};
 
 	pinctrl_usb_hub_rst: usbhubrstgrp {
 		fsl,pins = <
-			MX8MP_IOMUXC_SD1_RESET_B__GPIO2_IO10			0x19
+			MX8MP_IOMUXC_SD1_RESET_B__GPIO2_IO10			0x100
 		>;
 	};
 
@@ -149,7 +149,7 @@
 
 	pinctrl_usdhc2_gpio: usdhc2gpiogrp {
 		fsl,pins = <
-			MX8MP_IOMUXC_GPIO1_IO04__GPIO1_IO04			0x101
+			MX8MP_IOMUXC_GPIO1_IO04__GPIO1_IO04			0x100
 		>;
 	};
 };
diff --git a/arch/arm/dts/som-imx8m-plus.dts b/arch/arm/dts/som-imx8m-plus.dts
index e9440eca9ea..fa3c6a59ba7 100644
--- a/arch/arm/dts/som-imx8m-plus.dts
+++ b/arch/arm/dts/som-imx8m-plus.dts
@@ -456,50 +456,50 @@
 
 	pinctrl_hog: hoggrp {
 		fsl,pins = <
-			MX8MP_IOMUXC_GPIO1_IO11__GPIO1_IO11	0x19
-			MX8MP_IOMUXC_GPIO1_IO13__GPIO1_IO13	0x19
+			MX8MP_IOMUXC_GPIO1_IO11__GPIO1_IO11	0x16
+			MX8MP_IOMUXC_GPIO1_IO13__GPIO1_IO13	0x16
 		>;
 	};
 
 	pinctrl_eqos: eqosgrp {
 		fsl,pins = <
-			MX8MP_IOMUXC_ENET_MDC__ENET_QOS_MDC	0x3
-			MX8MP_IOMUXC_ENET_MDIO__ENET_QOS_MDIO	0x3
-			MX8MP_IOMUXC_ENET_RD0__ENET_QOS_RGMII_RD0	0x91
-			MX8MP_IOMUXC_ENET_RD1__ENET_QOS_RGMII_RD1	0x91
-			MX8MP_IOMUXC_ENET_RD2__ENET_QOS_RGMII_RD2	0x91
-			MX8MP_IOMUXC_ENET_RD3__ENET_QOS_RGMII_RD3	0x91
-			MX8MP_IOMUXC_ENET_RXC__CCM_ENET_QOS_CLOCK_GENERATE_RX_CLK	0x91
-			MX8MP_IOMUXC_ENET_RX_CTL__ENET_QOS_RGMII_RX_CTL	0x91
-			MX8MP_IOMUXC_ENET_TD0__ENET_QOS_RGMII_TD0		0x1f
-			MX8MP_IOMUXC_ENET_TD1__ENET_QOS_RGMII_TD1		0x1f
-			MX8MP_IOMUXC_ENET_TD2__ENET_QOS_RGMII_TD2		0x1f
-			MX8MP_IOMUXC_ENET_TD3__ENET_QOS_RGMII_TD3		0x1f
-			MX8MP_IOMUXC_ENET_TX_CTL__ENET_QOS_RGMII_TX_CTL	0x1f
-			MX8MP_IOMUXC_ENET_TXC__CCM_ENET_QOS_CLOCK_GENERATE_TX_CLK	0x1f
-			MX8MP_IOMUXC_GPIO1_IO10__GPIO1_IO10	0x19
+			MX8MP_IOMUXC_ENET_MDC__ENET_QOS_MDC	0x2
+			MX8MP_IOMUXC_ENET_MDIO__ENET_QOS_MDIO	0x2
+			MX8MP_IOMUXC_ENET_RD0__ENET_QOS_RGMII_RD0	0x90
+			MX8MP_IOMUXC_ENET_RD1__ENET_QOS_RGMII_RD1	0x90
+			MX8MP_IOMUXC_ENET_RD2__ENET_QOS_RGMII_RD2	0x90
+			MX8MP_IOMUXC_ENET_RD3__ENET_QOS_RGMII_RD3	0x90
+			MX8MP_IOMUXC_ENET_RXC__CCM_ENET_QOS_CLOCK_GENERATE_RX_CLK	0x90
+			MX8MP_IOMUXC_ENET_RX_CTL__ENET_QOS_RGMII_RX_CTL	0x90
+			MX8MP_IOMUXC_ENET_TD0__ENET_QOS_RGMII_TD0		0x16
+			MX8MP_IOMUXC_ENET_TD1__ENET_QOS_RGMII_TD1		0x16
+			MX8MP_IOMUXC_ENET_TD2__ENET_QOS_RGMII_TD2		0x16
+			MX8MP_IOMUXC_ENET_TD3__ENET_QOS_RGMII_TD3		0x16
+			MX8MP_IOMUXC_ENET_TX_CTL__ENET_QOS_RGMII_TX_CTL	0x16
+			MX8MP_IOMUXC_ENET_TXC__CCM_ENET_QOS_CLOCK_GENERATE_TX_CLK	0x16
+			MX8MP_IOMUXC_GPIO1_IO10__GPIO1_IO10	0x10
 		>;
 	};
 
 	pinctrl_fec: fecgrp {
 		fsl,pins = <
-			MX8MP_IOMUXC_SAI1_RXD2__ENET1_MDC		0x3
-			MX8MP_IOMUXC_SAI1_RXD3__ENET1_MDIO		0x3
-			MX8MP_IOMUXC_SAI1_RXD4__ENET1_RGMII_RD0		0x91
-			MX8MP_IOMUXC_SAI1_RXD5__ENET1_RGMII_RD1		0x91
-			MX8MP_IOMUXC_SAI1_RXD6__ENET1_RGMII_RD2		0x91
-			MX8MP_IOMUXC_SAI1_RXD7__ENET1_RGMII_RD3		0x91
-			MX8MP_IOMUXC_SAI1_TXC__ENET1_RGMII_RXC		0x91
-			MX8MP_IOMUXC_SAI1_TXFS__ENET1_RGMII_RX_CTL	0x91
-			MX8MP_IOMUXC_SAI1_TXD0__ENET1_RGMII_TD0		0x1f
-			MX8MP_IOMUXC_SAI1_TXD1__ENET1_RGMII_TD1		0x1f
-			MX8MP_IOMUXC_SAI1_TXD2__ENET1_RGMII_TD2		0x1f
-			MX8MP_IOMUXC_SAI1_TXD3__ENET1_RGMII_TD3		0x1f
-			MX8MP_IOMUXC_SAI1_TXD4__ENET1_RGMII_TX_CTL	0x1f
-			MX8MP_IOMUXC_SAI1_TXD5__ENET1_RGMII_TXC		0x1f
-			MX8MP_IOMUXC_GPIO1_IO05__GPIO1_IO05		0x19
-			MX8MP_IOMUXC_SAI2_TXFS__GPIO4_IO24		0x19
-			MX8MP_IOMUXC_SAI2_RXFS__GPIO4_IO21		0x19
+			MX8MP_IOMUXC_SAI1_RXD2__ENET1_MDC		0x2
+			MX8MP_IOMUXC_SAI1_RXD3__ENET1_MDIO		0x2
+			MX8MP_IOMUXC_SAI1_RXD4__ENET1_RGMII_RD0		0x90
+			MX8MP_IOMUXC_SAI1_RXD5__ENET1_RGMII_RD1		0x90
+			MX8MP_IOMUXC_SAI1_RXD6__ENET1_RGMII_RD2		0x90
+			MX8MP_IOMUXC_SAI1_RXD7__ENET1_RGMII_RD3		0x90
+			MX8MP_IOMUXC_SAI1_TXC__ENET1_RGMII_RXC		0x90
+			MX8MP_IOMUXC_SAI1_TXFS__ENET1_RGMII_RX_CTL	0x90
+			MX8MP_IOMUXC_SAI1_TXD0__ENET1_RGMII_TD0		0x16
+			MX8MP_IOMUXC_SAI1_TXD1__ENET1_RGMII_TD1		0x16
+			MX8MP_IOMUXC_SAI1_TXD2__ENET1_RGMII_TD2		0x16
+			MX8MP_IOMUXC_SAI1_TXD3__ENET1_RGMII_TD3		0x16
+			MX8MP_IOMUXC_SAI1_TXD4__ENET1_RGMII_TX_CTL	0x16
+			MX8MP_IOMUXC_SAI1_TXD5__ENET1_RGMII_TXC		0x16
+			MX8MP_IOMUXC_GPIO1_IO05__GPIO1_IO05		0x16
+			MX8MP_IOMUXC_SAI2_TXFS__GPIO4_IO24		0x16
+			MX8MP_IOMUXC_SAI2_RXFS__GPIO4_IO21		0x16
 		>;
 	};
 
@@ -516,83 +516,83 @@
 
 	pinctrl_gpio_led: gpioledgrp {
 		fsl,pins = <
-			MX8MP_IOMUXC_GPIO1_IO12__GPIO1_IO12	0x19
+			MX8MP_IOMUXC_GPIO1_IO12__GPIO1_IO12	0x140
 		>;
 	};
 
 	pinctrl_i2c1: i2c1grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C1_SCL__I2C1_SCL		0x400001c3
-			MX8MP_IOMUXC_I2C1_SDA__I2C1_SDA		0x400001c3
+			MX8MP_IOMUXC_I2C1_SCL__I2C1_SCL		0x400001c2
+			MX8MP_IOMUXC_I2C1_SDA__I2C1_SDA		0x400001c2
 		>;
 	};
 
 	pinctrl_i2c2: i2c2grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C2_SCL__I2C2_SCL			0x400001c3
-			MX8MP_IOMUXC_I2C2_SDA__I2C2_SDA			0x400001c3
+			MX8MP_IOMUXC_I2C2_SCL__I2C2_SCL			0x400001c2
+			MX8MP_IOMUXC_I2C2_SDA__I2C2_SDA			0x400001c2
 		>;
 	};
 
 	pinctrl_i2c3: i2c3grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C3_SCL__I2C3_SCL			0x400001c3
-			MX8MP_IOMUXC_I2C3_SDA__I2C3_SDA			0x400001c3
+			MX8MP_IOMUXC_I2C3_SCL__I2C3_SCL			0x400001c2
+			MX8MP_IOMUXC_I2C3_SDA__I2C3_SDA			0x400001c2
 		>;
 	};
 
 	pinctrl_i2c4: i2c4grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C4_SCL__I2C4_SCL			0x400001c3
-			MX8MP_IOMUXC_I2C4_SDA__I2C4_SDA			0x400001c3
+			MX8MP_IOMUXC_I2C4_SCL__I2C4_SCL			0x400001c2
+			MX8MP_IOMUXC_I2C4_SDA__I2C4_SDA			0x400001c2
 		>;
 	};
 
 	pinctrl_i2c5: i2c5grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_SAI5_MCLK__I2C5_SDA		0x400001c3
-			MX8MP_IOMUXC_SAI5_RXD0__I2C5_SCL		0x400001c3
+			MX8MP_IOMUXC_SAI5_MCLK__I2C5_SDA		0x400001c2
+			MX8MP_IOMUXC_SAI5_RXD0__I2C5_SCL		0x400001c2
 		>;
 	};
 
 	pinctrl_i2c1_gpio: i2c1grp-gpio {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C1_SCL__GPIO5_IO14        	0x1c3
-			MX8MP_IOMUXC_I2C1_SDA__GPIO5_IO15        	0x1c3
+			MX8MP_IOMUXC_I2C1_SCL__GPIO5_IO14        	0x1c2
+			MX8MP_IOMUXC_I2C1_SDA__GPIO5_IO15        	0x1c2
 		>;
 	};
 
 	pinctrl_i2c2_gpio: i2c2grp-gpio {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C2_SCL__GPIO5_IO16        	0x1c3
-			MX8MP_IOMUXC_I2C2_SDA__GPIO5_IO17        	0x1c3
+			MX8MP_IOMUXC_I2C2_SCL__GPIO5_IO16        	0x1c2
+			MX8MP_IOMUXC_I2C2_SDA__GPIO5_IO17        	0x1c2
 		>;
 	};
 
 	pinctrl_i2c3_gpio: i2c3grp-gpio {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C3_SCL__GPIO5_IO18        	0x1c3
-			MX8MP_IOMUXC_I2C3_SDA__GPIO5_IO19        	0x1c3
+			MX8MP_IOMUXC_I2C3_SCL__GPIO5_IO18        	0x1c2
+			MX8MP_IOMUXC_I2C3_SDA__GPIO5_IO19        	0x1c2
 		>;
 	};
 
 	pinctrl_i2c4_gpio: i2c4grp-gpio {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C4_SCL__GPIO5_IO20       	0x1c3
-			MX8MP_IOMUXC_I2C4_SDA__GPIO5_IO21        	0x1c3
+			MX8MP_IOMUXC_I2C4_SCL__GPIO5_IO20       	0x1c2
+			MX8MP_IOMUXC_I2C4_SDA__GPIO5_IO21        	0x1c2
 		>;
 	};
 
 	pinctrl_i2c5_gpio: i2c5grp-gpio {
 		fsl,pins = <
-			MX8MP_IOMUXC_SAI5_MCLK__GPIO3_IO25       	0x1c3
-			MX8MP_IOMUXC_SAI5_RXD0__GPIO3_IO21        	0x1c3
+			MX8MP_IOMUXC_SAI5_MCLK__GPIO3_IO25       	0x1c2
+			MX8MP_IOMUXC_SAI5_RXD0__GPIO3_IO21        	0x1c2
 		>;
 	};
 
 	pinctrl_pmic: pmicirq {
 		fsl,pins = <
-			MX8MP_IOMUXC_GPIO1_IO03__GPIO1_IO03	0x41
+			MX8MP_IOMUXC_GPIO1_IO03__GPIO1_IO03	0x000001c0
 		>;
 	};
 
@@ -610,26 +610,26 @@
 
 	pinctrl_reg_usdhc2_vmmc: regusdhc2vmmc {
 		fsl,pins = <
-			MX8MP_IOMUXC_SD2_RESET_B__GPIO2_IO19	0x41
+			MX8MP_IOMUXC_SD2_RESET_B__GPIO2_IO19	0x140
 		>;
 	};
 
 	pinctrl_uart2: uart2grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_UART2_RXD__UART2_DCE_RX	0x49
-			MX8MP_IOMUXC_UART2_TXD__UART2_DCE_TX	0x49
+			MX8MP_IOMUXC_UART2_RXD__UART2_DCE_RX	0x140
+			MX8MP_IOMUXC_UART2_TXD__UART2_DCE_TX	0x140
 		>;
 	};
 
 	pinctrl_usb1_vbus: usb1grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_GPIO1_IO14__GPIO1_IO14	0x19
+			MX8MP_IOMUXC_GPIO1_IO14__GPIO1_IO14	0x10
 		>;
 	};
 
 	pinctrl_3v3_exp: exp1grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_SD1_STROBE__GPIO2_IO11	0x19
+			MX8MP_IOMUXC_SD1_STROBE__GPIO2_IO11	0x10
 		>;
 	};
 
@@ -641,7 +641,7 @@
 			MX8MP_IOMUXC_SD2_DATA1__USDHC2_DATA1	0x1d0
 			MX8MP_IOMUXC_SD2_DATA2__USDHC2_DATA2	0x1d0
 			MX8MP_IOMUXC_SD2_DATA3__USDHC2_DATA3	0x1d0
-			MX8MP_IOMUXC_GPIO1_IO04__USDHC2_VSELECT	0xc1
+			MX8MP_IOMUXC_GPIO1_IO04__USDHC2_VSELECT	0xc0
 		>;
 	};
 
@@ -653,7 +653,7 @@
 			MX8MP_IOMUXC_SD2_DATA1__USDHC2_DATA1	0x1d4
 			MX8MP_IOMUXC_SD2_DATA2__USDHC2_DATA2	0x1d4
 			MX8MP_IOMUXC_SD2_DATA3__USDHC2_DATA3	0x1d4
-			MX8MP_IOMUXC_GPIO1_IO04__USDHC2_VSELECT 0xc1
+			MX8MP_IOMUXC_GPIO1_IO04__USDHC2_VSELECT 0xc0
 		>;
 	};
 
@@ -665,7 +665,7 @@
 			MX8MP_IOMUXC_SD2_DATA1__USDHC2_DATA1	0x1d6
 			MX8MP_IOMUXC_SD2_DATA2__USDHC2_DATA2	0x1d6
 			MX8MP_IOMUXC_SD2_DATA3__USDHC2_DATA3	0x1d6
-			MX8MP_IOMUXC_GPIO1_IO04__USDHC2_VSELECT 0xc1
+			MX8MP_IOMUXC_GPIO1_IO04__USDHC2_VSELECT 0xc0
 		>;
 	};
 
diff --git a/arch/arm/dts/ucm-imx8m-plus-pinctrl.dtsi b/arch/arm/dts/ucm-imx8m-plus-pinctrl.dtsi
index 581e57d9f28..2ee43e6d3da 100644
--- a/arch/arm/dts/ucm-imx8m-plus-pinctrl.dtsi
+++ b/arch/arm/dts/ucm-imx8m-plus-pinctrl.dtsi
@@ -6,40 +6,40 @@
 &iomuxc {
 	pinctrl_fec: fecgrp {
 		fsl,pins = <
-			MX8MP_IOMUXC_SD1_CLK__ENET1_MDC					0x3
-			MX8MP_IOMUXC_SD1_CMD__ENET1_MDIO				0x3
-			MX8MP_IOMUXC_SD1_DATA2__ENET1_RGMII_RD0				0x91
-			MX8MP_IOMUXC_SD1_DATA3__ENET1_RGMII_RD1				0x91
-			MX8MP_IOMUXC_SAI1_RXD6__ENET1_RGMII_RD2				0x91
-			MX8MP_IOMUXC_SAI1_RXD7__ENET1_RGMII_RD3				0x91
-			MX8MP_IOMUXC_SAI1_TXC__ENET1_RGMII_RXC				0x91
-			MX8MP_IOMUXC_SAI1_TXFS__ENET1_RGMII_RX_CTL			0x91
-			MX8MP_IOMUXC_SD1_DATA1__ENET1_RGMII_TD0				0x1f
-			MX8MP_IOMUXC_SD1_DATA0__ENET1_RGMII_TD1				0x1f
-			MX8MP_IOMUXC_SAI1_TXD2__ENET1_RGMII_TD2				0x1f
-			MX8MP_IOMUXC_SAI1_TXD3__ENET1_RGMII_TD3				0x1f
-			MX8MP_IOMUXC_SAI1_TXD4__ENET1_RGMII_TX_CTL			0x1f
-			MX8MP_IOMUXC_SAI1_TXD5__ENET1_RGMII_TXC				0x1f
+			MX8MP_IOMUXC_SD1_CLK__ENET1_MDC					0x2
+			MX8MP_IOMUXC_SD1_CMD__ENET1_MDIO				0x2
+			MX8MP_IOMUXC_SD1_DATA2__ENET1_RGMII_RD0				0x90
+			MX8MP_IOMUXC_SD1_DATA3__ENET1_RGMII_RD1				0x90
+			MX8MP_IOMUXC_SAI1_RXD6__ENET1_RGMII_RD2				0x90
+			MX8MP_IOMUXC_SAI1_RXD7__ENET1_RGMII_RD3				0x90
+			MX8MP_IOMUXC_SAI1_TXC__ENET1_RGMII_RXC				0x90
+			MX8MP_IOMUXC_SAI1_TXFS__ENET1_RGMII_RX_CTL			0x90
+			MX8MP_IOMUXC_SD1_DATA1__ENET1_RGMII_TD0				0x16
+			MX8MP_IOMUXC_SD1_DATA0__ENET1_RGMII_TD1				0x16
+			MX8MP_IOMUXC_SAI1_TXD2__ENET1_RGMII_TD2				0x16
+			MX8MP_IOMUXC_SAI1_TXD3__ENET1_RGMII_TD3				0x16
+			MX8MP_IOMUXC_SAI1_TXD4__ENET1_RGMII_TX_CTL			0x16
+			MX8MP_IOMUXC_SAI1_TXD5__ENET1_RGMII_TXC				0x16
 		>;
 	};
     
 	pinctrl_eqos: eqosgrp {
 		fsl,pins = <
-			MX8MP_IOMUXC_ENET_MDC__ENET_QOS_MDC				0x3
-			MX8MP_IOMUXC_ENET_MDIO__ENET_QOS_MDIO				0x3
-			MX8MP_IOMUXC_ENET_RD0__ENET_QOS_RGMII_RD0			0x91
-			MX8MP_IOMUXC_ENET_RD1__ENET_QOS_RGMII_RD1			0x91
-			MX8MP_IOMUXC_ENET_RD2__ENET_QOS_RGMII_RD2			0x91
-			MX8MP_IOMUXC_ENET_RD3__ENET_QOS_RGMII_RD3			0x91
-			MX8MP_IOMUXC_ENET_RXC__CCM_ENET_QOS_CLOCK_GENERATE_RX_CLK	0x91
-			MX8MP_IOMUXC_ENET_RX_CTL__ENET_QOS_RGMII_RX_CTL			0x91
-			MX8MP_IOMUXC_ENET_TD0__ENET_QOS_RGMII_TD0			0x1f
-			MX8MP_IOMUXC_ENET_TD1__ENET_QOS_RGMII_TD1			0x1f
-			MX8MP_IOMUXC_ENET_TD2__ENET_QOS_RGMII_TD2			0x1f
-			MX8MP_IOMUXC_ENET_TD3__ENET_QOS_RGMII_TD3			0x1f
-			MX8MP_IOMUXC_ENET_TX_CTL__ENET_QOS_RGMII_TX_CTL			0x1f
-			MX8MP_IOMUXC_ENET_TXC__CCM_ENET_QOS_CLOCK_GENERATE_TX_CLK	0x1f
-			MX8MP_IOMUXC_GPIO1_IO10__GPIO1_IO10				0x19
+			MX8MP_IOMUXC_ENET_MDC__ENET_QOS_MDC				0x2
+			MX8MP_IOMUXC_ENET_MDIO__ENET_QOS_MDIO				0x2
+			MX8MP_IOMUXC_ENET_RD0__ENET_QOS_RGMII_RD0			0x90
+			MX8MP_IOMUXC_ENET_RD1__ENET_QOS_RGMII_RD1			0x90
+			MX8MP_IOMUXC_ENET_RD2__ENET_QOS_RGMII_RD2			0x90
+			MX8MP_IOMUXC_ENET_RD3__ENET_QOS_RGMII_RD3			0x90
+			MX8MP_IOMUXC_ENET_RXC__CCM_ENET_QOS_CLOCK_GENERATE_RX_CLK	0x90
+			MX8MP_IOMUXC_ENET_RX_CTL__ENET_QOS_RGMII_RX_CTL			0x90
+			MX8MP_IOMUXC_ENET_TD0__ENET_QOS_RGMII_TD0			0x16
+			MX8MP_IOMUXC_ENET_TD1__ENET_QOS_RGMII_TD1			0x16
+			MX8MP_IOMUXC_ENET_TD2__ENET_QOS_RGMII_TD2			0x16
+			MX8MP_IOMUXC_ENET_TD3__ENET_QOS_RGMII_TD3			0x16
+			MX8MP_IOMUXC_ENET_TX_CTL__ENET_QOS_RGMII_TX_CTL			0x16
+			MX8MP_IOMUXC_ENET_TXC__CCM_ENET_QOS_CLOCK_GENERATE_TX_CLK	0x16
+			MX8MP_IOMUXC_GPIO1_IO10__GPIO1_IO10				0x10
 		>;
 	};
 
@@ -56,83 +56,83 @@
 
 	pinctrl_gpio_led: gpioledgrp {
 		fsl,pins = <
-			MX8MP_IOMUXC_GPIO1_IO12__GPIO1_IO12				0x19
+			MX8MP_IOMUXC_GPIO1_IO12__GPIO1_IO12				0x140
 		>;
 	};
 
 	pinctrl_i2c1: i2c1grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C1_SCL__I2C1_SCL					0x400001c3
-			MX8MP_IOMUXC_I2C1_SDA__I2C1_SDA					0x400001c3
+			MX8MP_IOMUXC_I2C1_SCL__I2C1_SCL					0x400001c2
+			MX8MP_IOMUXC_I2C1_SDA__I2C1_SDA					0x400001c2
 		>;
 	};
 
 	pinctrl_i2c2: i2c2grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C2_SCL__I2C2_SCL					0x400001c3
-			MX8MP_IOMUXC_I2C2_SDA__I2C2_SDA					0x400001c3
+			MX8MP_IOMUXC_I2C2_SCL__I2C2_SCL					0x400001c2
+			MX8MP_IOMUXC_I2C2_SDA__I2C2_SDA					0x400001c2
 		>;
 	};
 
 	pinctrl_i2c3: i2c3grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C3_SCL__I2C3_SCL					0x400001c3
-			MX8MP_IOMUXC_I2C3_SDA__I2C3_SDA					0x400001c3
+			MX8MP_IOMUXC_I2C3_SCL__I2C3_SCL					0x400001c2
+			MX8MP_IOMUXC_I2C3_SDA__I2C3_SDA					0x400001c2
 		>;
 	};
 
 	pinctrl_i2c5: i2c5grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_SAI5_MCLK__I2C5_SDA				0x400001c3
-			MX8MP_IOMUXC_SAI5_RXD0__I2C5_SCL				0x400001c3
+			MX8MP_IOMUXC_SAI5_MCLK__I2C5_SDA				0x400001c2
+			MX8MP_IOMUXC_SAI5_RXD0__I2C5_SCL				0x400001c2
 		>;
 	};
 
 	pinctrl_i2c1_gpio: i2c1grp-gpio {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C1_SCL__GPIO5_IO14				0x1c3
-			MX8MP_IOMUXC_I2C1_SDA__GPIO5_IO15				0x1c3
+			MX8MP_IOMUXC_I2C1_SCL__GPIO5_IO14				0x1c2
+			MX8MP_IOMUXC_I2C1_SDA__GPIO5_IO15				0x1c2
 		>;
 	};
 
 	pinctrl_i2c2_gpio: i2c2grp-gpio {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C2_SCL__GPIO5_IO16				0x1c3
-			MX8MP_IOMUXC_I2C2_SDA__GPIO5_IO17				0x1c3
+			MX8MP_IOMUXC_I2C2_SCL__GPIO5_IO16				0x1c2
+			MX8MP_IOMUXC_I2C2_SDA__GPIO5_IO17				0x1c2
 		>;
 	};
 
 	pinctrl_i2c3_gpio: i2c3grp-gpio {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C3_SCL__GPIO5_IO18				0x1c3
-			MX8MP_IOMUXC_I2C3_SDA__GPIO5_IO19				0x1c3
+			MX8MP_IOMUXC_I2C3_SCL__GPIO5_IO18				0x1c2
+			MX8MP_IOMUXC_I2C3_SDA__GPIO5_IO19				0x1c2
 		>;
 	};
 
 	pinctrl_i2c5_gpio: i2c5grp-gpio {
 		fsl,pins = <
-			MX8MP_IOMUXC_SAI5_MCLK__GPIO3_IO25				0x1c3
-			MX8MP_IOMUXC_SAI5_RXD0__GPIO3_IO21				0x1c3
+			MX8MP_IOMUXC_SAI5_MCLK__GPIO3_IO25				0x1c2
+			MX8MP_IOMUXC_SAI5_RXD0__GPIO3_IO21				0x1c2
 		>;
 	};
 
 	pinctrl_pmic: pmicirq {
 		fsl,pins = <
-			MX8MP_IOMUXC_GPIO1_IO03__GPIO1_IO03				0x41
+			MX8MP_IOMUXC_GPIO1_IO03__GPIO1_IO03				0x000001c0
 		>;
 	};
 
 	pinctrl_uart1: uart1grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_UART1_RXD__UART1_DCE_RX				0x49
-			MX8MP_IOMUXC_UART1_TXD__UART1_DCE_TX				0x49
+			MX8MP_IOMUXC_UART1_RXD__UART1_DCE_RX				0x140
+			MX8MP_IOMUXC_UART1_TXD__UART1_DCE_TX				0x140
 		>;
 	};
 
 	pinctrl_uart2: uart2grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_UART2_RXD__UART2_DCE_RX				0x49
-			MX8MP_IOMUXC_UART2_TXD__UART2_DCE_TX				0x49
+			MX8MP_IOMUXC_UART2_RXD__UART2_DCE_RX				0x140
+			MX8MP_IOMUXC_UART2_TXD__UART2_DCE_TX				0x140
 		>;
 	};
 
diff --git a/arch/arm/dts/ucm-imx8m-plus.dts b/arch/arm/dts/ucm-imx8m-plus.dts
index 70cede75615..0c9c6afa259 100644
--- a/arch/arm/dts/ucm-imx8m-plus.dts
+++ b/arch/arm/dts/ucm-imx8m-plus.dts
@@ -503,8 +503,8 @@
 
 	pinctrl_hog: hoggrp {
 		fsl,pins = <
-			MX8MP_IOMUXC_GPIO1_IO11__GPIO1_IO11	0x19
-			MX8MP_IOMUXC_GPIO1_IO13__GPIO1_IO13	0x19
+			MX8MP_IOMUXC_GPIO1_IO11__GPIO1_IO11	0x10
+			MX8MP_IOMUXC_GPIO1_IO13__GPIO1_IO13	0x10
 		>;
 	};
 
@@ -534,21 +534,21 @@
 
 	pinctrl_eqos: eqosgrp {
 		fsl,pins = <
-			MX8MP_IOMUXC_ENET_MDC__ENET_QOS_MDC	0x3
-			MX8MP_IOMUXC_ENET_MDIO__ENET_QOS_MDIO	0x3
-			MX8MP_IOMUXC_ENET_RD0__ENET_QOS_RGMII_RD0	0x91
-			MX8MP_IOMUXC_ENET_RD1__ENET_QOS_RGMII_RD1	0x91
-			MX8MP_IOMUXC_ENET_RD2__ENET_QOS_RGMII_RD2	0x91
-			MX8MP_IOMUXC_ENET_RD3__ENET_QOS_RGMII_RD3	0x91
-			MX8MP_IOMUXC_ENET_RXC__CCM_ENET_QOS_CLOCK_GENERATE_RX_CLK	0x91
-			MX8MP_IOMUXC_ENET_RX_CTL__ENET_QOS_RGMII_RX_CTL	0x91
-			MX8MP_IOMUXC_ENET_TD0__ENET_QOS_RGMII_TD0		0x1f
-			MX8MP_IOMUXC_ENET_TD1__ENET_QOS_RGMII_TD1		0x1f
-			MX8MP_IOMUXC_ENET_TD2__ENET_QOS_RGMII_TD2		0x1f
-			MX8MP_IOMUXC_ENET_TD3__ENET_QOS_RGMII_TD3		0x1f
-			MX8MP_IOMUXC_ENET_TX_CTL__ENET_QOS_RGMII_TX_CTL	0x1f
-			MX8MP_IOMUXC_ENET_TXC__CCM_ENET_QOS_CLOCK_GENERATE_TX_CLK	0x1f
-			MX8MP_IOMUXC_GPIO1_IO10__GPIO1_IO10	0x19
+			MX8MP_IOMUXC_ENET_MDC__ENET_QOS_MDC	0x2
+			MX8MP_IOMUXC_ENET_MDIO__ENET_QOS_MDIO	0x2
+			MX8MP_IOMUXC_ENET_RD0__ENET_QOS_RGMII_RD0	0x90
+			MX8MP_IOMUXC_ENET_RD1__ENET_QOS_RGMII_RD1	0x90
+			MX8MP_IOMUXC_ENET_RD2__ENET_QOS_RGMII_RD2	0x90
+			MX8MP_IOMUXC_ENET_RD3__ENET_QOS_RGMII_RD3	0x90
+			MX8MP_IOMUXC_ENET_RXC__CCM_ENET_QOS_CLOCK_GENERATE_RX_CLK	0x90
+			MX8MP_IOMUXC_ENET_RX_CTL__ENET_QOS_RGMII_RX_CTL	0x90
+			MX8MP_IOMUXC_ENET_TD0__ENET_QOS_RGMII_TD0		0x16
+			MX8MP_IOMUXC_ENET_TD1__ENET_QOS_RGMII_TD1		0x16
+			MX8MP_IOMUXC_ENET_TD2__ENET_QOS_RGMII_TD2		0x16
+			MX8MP_IOMUXC_ENET_TD3__ENET_QOS_RGMII_TD3		0x16
+			MX8MP_IOMUXC_ENET_TX_CTL__ENET_QOS_RGMII_TX_CTL	0x16
+			MX8MP_IOMUXC_ENET_TXC__CCM_ENET_QOS_CLOCK_GENERATE_TX_CLK	0x16
+			MX8MP_IOMUXC_GPIO1_IO10__GPIO1_IO10	0x10
 		>;
 	};
 
@@ -565,69 +565,69 @@
 
 	pinctrl_gpio_led: gpioledgrp {
 		fsl,pins = <
-			MX8MP_IOMUXC_GPIO1_IO12__GPIO1_IO12	0x19
+			MX8MP_IOMUXC_GPIO1_IO12__GPIO1_IO12	0x140
 		>;
 	};
 
 	pinctrl_i2c1: i2c1grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C1_SCL__I2C1_SCL		0x400001c3
-			MX8MP_IOMUXC_I2C1_SDA__I2C1_SDA		0x400001c3
+			MX8MP_IOMUXC_I2C1_SCL__I2C1_SCL		0x400001c2
+			MX8MP_IOMUXC_I2C1_SDA__I2C1_SDA		0x400001c2
 		>;
 	};
 
 	pinctrl_i2c2: i2c2grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C2_SCL__I2C2_SCL			0x400001c3
-			MX8MP_IOMUXC_I2C2_SDA__I2C2_SDA			0x400001c3
+			MX8MP_IOMUXC_I2C2_SCL__I2C2_SCL			0x400001c2
+			MX8MP_IOMUXC_I2C2_SDA__I2C2_SDA			0x400001c2
 		>;
 	};
 
 	pinctrl_i2c3: i2c3grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C3_SCL__I2C3_SCL			0x400001c3
-			MX8MP_IOMUXC_I2C3_SDA__I2C3_SDA			0x400001c3
+			MX8MP_IOMUXC_I2C3_SCL__I2C3_SCL			0x400001c2
+			MX8MP_IOMUXC_I2C3_SDA__I2C3_SDA			0x400001c2
 		>;
 	};
 
 	pinctrl_i2c5: i2c5grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_SAI5_MCLK__I2C5_SDA		0x400001c3
-			MX8MP_IOMUXC_SAI5_RXD0__I2C5_SCL		0x400001c3
+			MX8MP_IOMUXC_SAI5_MCLK__I2C5_SDA		0x400001c2
+			MX8MP_IOMUXC_SAI5_RXD0__I2C5_SCL		0x400001c2
 		>;
 	};
 
 	pinctrl_i2c1_gpio: i2c1grp-gpio {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C1_SCL__GPIO5_IO14        	0x1c3
-			MX8MP_IOMUXC_I2C1_SDA__GPIO5_IO15        	0x1c3
+			MX8MP_IOMUXC_I2C1_SCL__GPIO5_IO14        	0x1c2
+			MX8MP_IOMUXC_I2C1_SDA__GPIO5_IO15        	0x1c2
 		>;
 	};
 
 	pinctrl_i2c2_gpio: i2c2grp-gpio {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C2_SCL__GPIO5_IO16        	0x1c3
-			MX8MP_IOMUXC_I2C2_SDA__GPIO5_IO17        	0x1c3
+			MX8MP_IOMUXC_I2C2_SCL__GPIO5_IO16        	0x1c2
+			MX8MP_IOMUXC_I2C2_SDA__GPIO5_IO17        	0x1c2
 		>;
 	};
 
 	pinctrl_i2c3_gpio: i2c3grp-gpio {
 		fsl,pins = <
-			MX8MP_IOMUXC_I2C3_SCL__GPIO5_IO18        	0x1c3
-			MX8MP_IOMUXC_I2C3_SDA__GPIO5_IO19        	0x1c3
+			MX8MP_IOMUXC_I2C3_SCL__GPIO5_IO18        	0x1c2
+			MX8MP_IOMUXC_I2C3_SDA__GPIO5_IO19        	0x1c2
 		>;
 	};
 
 	pinctrl_i2c5_gpio: i2c5grp-gpio {
 		fsl,pins = <
-			MX8MP_IOMUXC_SAI5_MCLK__GPIO3_IO25       	0x1c3
-			MX8MP_IOMUXC_SAI5_RXD0__GPIO3_IO21        	0x1c3
+			MX8MP_IOMUXC_SAI5_MCLK__GPIO3_IO25       	0x1c2
+			MX8MP_IOMUXC_SAI5_RXD0__GPIO3_IO21        	0x1c2
 		>;
 	};
 
 	pinctrl_pmic: pmicirq {
 		fsl,pins = <
-			MX8MP_IOMUXC_GPIO1_IO03__GPIO1_IO03	0x41
+			MX8MP_IOMUXC_GPIO1_IO03__GPIO1_IO03	0x000001c0
 		>;
 	};
 
@@ -645,21 +645,21 @@
 
 	pinctrl_reg_usdhc2_vmmc: regusdhc2vmmc {
 		fsl,pins = <
-			MX8MP_IOMUXC_SD2_RESET_B__GPIO2_IO19	0x41
+			MX8MP_IOMUXC_SD2_RESET_B__GPIO2_IO19	0x140
 		>;
 	};
 
 	pinctrl_uart1: uart1grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_SD1_CMD__UART1_DCE_RX	0x49
-			MX8MP_IOMUXC_SAI2_RXFS__UART1_DCE_TX	0x49
+			MX8MP_IOMUXC_SD1_CMD__UART1_DCE_RX	0x140
+			MX8MP_IOMUXC_SAI2_RXFS__UART1_DCE_TX	0x140
 		>;
 	};
 
 	pinctrl_uart2: uart2grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_UART2_RXD__UART2_DCE_RX	0x49
-			MX8MP_IOMUXC_UART2_TXD__UART2_DCE_TX	0x49
+			MX8MP_IOMUXC_UART2_RXD__UART2_DCE_RX	0x140
+			MX8MP_IOMUXC_UART2_TXD__UART2_DCE_TX	0x140
 		>;
 	};
 
@@ -677,7 +677,7 @@
 			MX8MP_IOMUXC_SD2_DATA1__USDHC2_DATA1	0x1d0
 			MX8MP_IOMUXC_SD2_DATA2__USDHC2_DATA2	0x1d0
 			MX8MP_IOMUXC_SD2_DATA3__USDHC2_DATA3	0x1d0
-			MX8MP_IOMUXC_GPIO1_IO04__USDHC2_VSELECT	0xc1
+			MX8MP_IOMUXC_GPIO1_IO04__USDHC2_VSELECT	0xc0
 		>;
 	};
 
@@ -689,7 +689,7 @@
 			MX8MP_IOMUXC_SD2_DATA1__USDHC2_DATA1	0x1d4
 			MX8MP_IOMUXC_SD2_DATA2__USDHC2_DATA2	0x1d4
 			MX8MP_IOMUXC_SD2_DATA3__USDHC2_DATA3	0x1d4
-			MX8MP_IOMUXC_GPIO1_IO04__USDHC2_VSELECT 0xc1
+			MX8MP_IOMUXC_GPIO1_IO04__USDHC2_VSELECT 0xc0
 		>;
 	};
 
@@ -701,7 +701,7 @@
 			MX8MP_IOMUXC_SD2_DATA1__USDHC2_DATA1	0x1d6
 			MX8MP_IOMUXC_SD2_DATA2__USDHC2_DATA2	0x1d6
 			MX8MP_IOMUXC_SD2_DATA3__USDHC2_DATA3	0x1d6
-			MX8MP_IOMUXC_GPIO1_IO04__USDHC2_VSELECT 0xc1
+			MX8MP_IOMUXC_GPIO1_IO04__USDHC2_VSELECT 0xc0
 		>;
 	};
 
-- 
2.34.1

